src/common/commonutils/CommonUtils.h

```diff
@@ -39,27 +42,45 @@
     }\
 }\
 
-#define OsConfigCaptureReason(reason, FORMAT1, FORMAT2, ...) {\
-    char* temp = NULL;\
+#define InternalOsConfigAddReason(reason, format, ...) {\
+    char* last = NULL;\
+    char* temp = FormatAllocateString("%s, also ", *reason);\
+    FREE_MEMORY(*reason);\

```

Comment:
Is it OK if `reason` is null here?

Reply:
I guess so, if only called by `OsConfigCaptureReason` which does the check?

Reply:
Yes. FREE_MEMORY (which I added and is here since beginning) checks for valid pointer before attempting free.

Reply:
Great!